lib/charms/mysql/v0/backups.py

```diff
@@ -533,22 +543,29 @@ def _on_restore(self, event: ActionEvent) -> None:
                 "completed": "ok",
             }
         )
+        # update status as soon as possible
+        self.charm._on_update_status(None)
 
-    def _pre_restore(self) -> Tuple[bool, Optional[str]]:
+    def _pre_restore(self) -> Tuple[bool, str]:
         """Perform operations that need to be done before performing a restore.
 
         Returns: tuple of (success, error_message)
         """
+        if not self.charm._mysql.is_mysqld_running():
+            return True, ""
+
         try:
+            logger.info("Stopping mysqld before restoring the backup")
```

Comment:
Should we remove `endpoint` from relation -> informing router/clients DB is not reachable?